src/main/java/org/ibp/api/brapi/v2/SampleResourceBrapi.java

```diff
@@ -5,79 +5,173 @@
+	@RequestMapping(value = "/{crop}/brapi/v2/samples", method = RequestMethod.GET)
+	@JsonView(BrapiView.BrapiV2.class)
+	@ResponseBody
+	public ResponseEntity<EntityListResponse<SampleObservationDto>> getSamples(@PathVariable final String crop,
+		@ApiParam(value = "the internal DB id for a sample")
+		@RequestParam(value = "sampleDbId", required = false) final String sampleDbId,
+		@ApiParam(value = "the internal DB id for an observation unit where a sample was taken from")
+		@RequestParam(value = "observationUnitDbId", required = false) final String observationUnitDbId,
+		@ApiParam(value = "the internal DB id for a plate of samples")
+		@RequestParam(value = "plateDbId", required = false) final String plateDbId,
+		@ApiParam(value = "the internal DB id for a germplasm")
+		@RequestParam(value = "germplasmDbId", required = false) final String germplasmDbId,
+		@ApiParam(value = "Filter by study DbId")
+		@RequestParam(value = "studyDbId", required = false) final String studyDbId,
+		@ApiParam(value = "An external reference ID. Could be a simple string or a URI. (use with externalReferenceSource parameter)")
+		@RequestParam(value = "externalReferenceID", required = false) final String externalReferenceID,
+		@ApiParam(value = "An identifier for the source system or database of an external reference (use with externalReferenceID parameter)")
+		@RequestParam(value = "externalReferenceSource", required = false) final String externalReferenceSource,
+		@ApiParam(value = BrapiPagedResult.CURRENT_PAGE_DESCRIPTION, required = false)
+		@RequestParam(value = "page", required = false) final Integer currentPage,
+		@ApiParam(value = BrapiPagedResult.PAGE_SIZE_DESCRIPTION, required = false)
+		@RequestParam(value = "pageSize", required = false) final Integer pageSize) {
+
+		final SampleSearchRequestDTO requestDTO = new SampleSearchRequestDTO(sampleDbId, observationUnitDbId, plateDbId,
+			germplasmDbId, studyDbId, externalReferenceID, externalReferenceSource);
+
+		final int finalPageNumber = currentPage == null ? BrapiPagedResult.DEFAULT_PAGE_NUMBER : currentPage;
+		final int finalPageSize = pageSize == null ? BrapiPagedResult.DEFAULT_PAGE_SIZE : pageSize;
+		final PageRequest pageRequest = new PageRequest(finalPageNumber, finalPageSize);
+		final PagedResult<SampleObservationDto> resultPage =
+			new PaginatedSearch().executeBrapiSearch(finalPageNumber, finalPageSize, new SearchSpec<SampleObservationDto>() {
+
+				@Override
+				public long getCount() {
+					return SampleResourceBrapi.this.sampleServiceBrapi.countSampleObservations(requestDTO);
+				}
+
+				@Override
+				public List<SampleObservationDto> getResults(final PagedResult<SampleObservationDto> pagedResult) {
+					return SampleResourceBrapi.this.sampleServiceBrapi.getSampleObservations(requestDTO, pageRequest);
+				}
+			});
+
+		final Result<SampleObservationDto> results = new Result<SampleObservationDto>().withData(resultPage.getPageResults());
```

Comment:
Extract to common method:
`final Result<SampleObservationDto> results = new Result<SampleObservationDto>().withData(resultPage.getPageResults());
		final Pagination pagination = new Pagination().withPageNumber(resultPage.getPageNumber()).withPageSize(resultPage.getPageSize())
			.withTotalCount(resultPage.getTotalResults()).withTotalPages(resultPage.getTotalPages());

		final Metadata metadata = new Metadata().withPagination(pagination);
		return new ResponseEntity<>(new EntityListResponse<>(metadata, results), HttpStatus.OK);`